### ISSUE TITLE
USB device (zwave) not usable in HA even though paths are detected by supervisor

### ISSUE THREAD

**Issue by kpine** [home-assistant/supervisor#1645]
**Home Assistant release with the issue:**
0.108.0

**Operating environment (HassOS/Generic):**
HassOS (supervised HA)

**Supervisor logs:**

[su_logs.txt](https://github.com/home-assistant/supervisor/files/4451760/su_logs.txt)

**Description of problem:**
If a USB device is inserted while HA is running, the paths are detected and visible, but HA cannot use them. For example, HA was up and running and I inserted a Z-Wave USB stick. After insertion the paths are shown. Here is the hardware info:
```
    serial:
        /dev/ttyS0
        /dev/ttyS2
        /dev/ttyACM0
        /dev/ttyS1
        /dev/ttyS3
        /dev/serial/by-id/usb-0658_0200_E2061B02-4A02-0114-3A06-FD1871291660-if00
    input:
        VirtualBox USB Tablet
        Sleep Button
        AT Translated Set 2 keyboard
        Video Bus
        ImExPS/2 Generic Explorer Mouse
        Power Button
    disk:
    gpio:
    audio:
```

`/dev/ttyACM0` (or the serial/by-id one) is the Z-Wave USB stick. If I then proceed to try and add the Z-Wave integration, I am presented with an error:

![image](https://user-images.githubusercontent.com/791794/78811805-4560ca00-797f-11ea-9857-54f6df751a9e.png)
**NOTE** I in fact used `/dev/ttyACM0` but the frontend reverts the text box after the error is returned. Because the zwave component does not log the actual exception it has caught, I copied over a zwave custom component with a modified `config_flow.py` which now logs the exception to the HA logs. Here is what it prints:
```
2020-04-08 09:56:40 ERROR (MainThread) [custom_components.zwave.config_flow] Failed to open USB path (/dev/ttyACM0): 'Zwave Generic Exception : Error when retrieving device /dev/ttyACM0 : [\'Traceback (most recent call last):\\n\', \'  File "/usr/local/lib/python3.7/site-packages/openzwave/option.py", line 78, in __init__\\n    raise ZWaveException(u"Can\\\'t find device %s : %s" % (device, traceback.format_exception(*sys.exc_info())))\\n\', \'openzwave.object.ZWaveException: "Zwave Generic Exception : Can\\\'t find device /dev/ttyACM0 : [\\\'NoneType: None\\\\\\\\n\\\']"\\n\']'
```
Basically OZW is unable to open the `/dev/ttyACM0`, even though the Supervisor says it is available. If you restart HA (Configuration -> Server Controls -> Restart) and try again it will add successfully. Seems to be some sort of discrepancy between what the supervisor sees and what the HA container is seeing?

This issue was seen by a user in the zwave Discord, along with another user in the community forums (posted [here](https://community.home-assistant.io/t/zwave-integration-with-docker/184869/)). The discord user was using the hassos image install on an RPi. The forum user is using supervised HA installed on Ubuntu. I reproduced myself using a VirtualBox VM using the provided VDI file. In all cases simply restarting HA solved i.t

**Comment by pvizeli** [comment]
Yeah, that is a docker issue. You can't map new device realtime into the container, you need restart it

**Comment by kpine** [comment]
It [is possible](https://www.balena.io/docs/reference/base-images/base-images/#working-with-dynamically-plugged-devices) to support dynamically plugged devices in containers, but maybe that approach doesn't work with HA. I had noticed the HA docker container was updated to now run the udev daemon, so I thought it might be related, but looks like no.

Anyways, I don't really any attachment to this issue, it was reported by two users on nearly the same day, so I thought I'd submit the issue. Perhaps it's a documentation or education concern then. Feel free to close this issue if there are no plans to address it.
